src/mpi/request/mpir_request.c

```diff
@@ -168,6 +168,15 @@ int MPIR_Request_completion_processing(MPIR_Request * request_ptr, MPI_Status *
                 break;
             }
 
+        case MPIR_REQUEST_KIND__PART_SEND:
+        case MPIR_REQUEST_KIND__PART_RECV:
+            {
+                MPID_Part_request_complete_hook(request_ptr);
+
+                MPIR_Request_extract_status(request_ptr, status);
+                mpi_errno = request_ptr->status.MPI_ERROR;
```

Comment:
I am also surprised that the completion process of partitioned request is different from persistent request. Can you clarify the difference, specifically, why we can't simply reuse the persistent request path here?

Reply:
The persistent pt2pt relies on an internal `real_request` which was created when sending or receiving the message at `MPID_Start` by calling `MPID_Irecv|Isend`. It does not expose the chance for NM or SHM native optimization. Thus, when completing a persistent request, it always frees `real_request` at MPIR.

We don't want to reuse the model of persistent calls for partitioned pt2pt because we allow NM or SHM to overwrite with different data transfer approaches. Currently basic version creates only a single AM req to track data transfer, but when we parallelize each partition transfer in future, we may need multiple reqs to keep track on concurrent data transfers. Thus, I don't think it is a good idea to force the same request completion process at MPIR for partitioned pt2pt.